### PR TITLE
Feature - Added CreateLinkPage

### PR DESCRIPTION
**Pull request by krishan711**
Screen Shot 2021-01-27 at 13 36 31

### CODE REVIEW

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      setPasswordError('Enter a valid target');
+      hasErrors = true;
+    }
+    if (!hasErrors) {
+      setIsLoading(true);
+      try {
+        await asyncSleep(1000);
+        navigator.navigateTo(`/domains/${props.domainId}`);
+      } catch (error) {
+        // TODO(rikhil): error to be rendered when login fails
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx:60]: yikes :P writing todos for me haha

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  };
+
+  const onPathTyped = (typedPath: string): void => {
+    if (pathError) {
+      setPathError(null);
+    }
+    setPath(typedPath);
+  };
+
+  const onPasswordTyped = (typedPassword: string): void => {
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: rename this?

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+export interface ICreateLinkPageProps {
+  domainId: string;
+}
+
+export const CreateLinkPage = (props: ICreateLinkPageProps): React.ReactElement => {
+  const navigator = useNavigator();
+  const [domain, setDomain] = React.useState<Domain | null>(null);
+  const [path, setPath] = React.useState<string | null>(null);
+  const [pathError, setPathError] = React.useState<string | null>(null);
+  const [target, setPassword] = React.useState<string | null>(null);
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: setTarget

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  const [targetError, setPasswordError] = React.useState<string | null>(null);
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: setTargetError

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  const loadDomain = (domainId: string): void => {
+    asyncSleep(300).then((): void => {
+      setDomain(DOMAIN_ID_MAP[domainId]);
+    }).catch((error: KibaException): void => {
+      console.error('error', error);
+      setDomain(null);
+    });
+  };
+
+  const isPathValid = (input: string): boolean => {
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: shall we move these to string util? likely won't be used again in divergent

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    const pattern = new RegExp('^'
+      + '(\\/[-a-z\\d%_.~+]*)*',
+    'i');
+    return !!pattern.test(input);
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: i think test returns a boolean so `!!` seems redundant

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  };
+
+  const isUrlValid = (input: string): boolean => {
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx]: can we use this for domain validation on the create domain page?

File: app/src/pages/createLinkPage/createLinkPage.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+import { Alignment, Box, Button, Direction, Form, InputType, PaddingSize, ResponsiveContainingView, SingleLineInput, Spacing, Stack, Text, TextAlignment } from '@kibalabs/ui-react';
+
+import { Domain, DOMAIN_ID_MAP } from '../../model';
+import { asyncSleep } from '../../util';
+
+export interface ICreateLinkPageProps {
+  domainId: string;
+}
+
+export const CreateLinkPage = (props: ICreateLinkPageProps): React.ReactElement => {
```
Comment (rikhilrai) [inline app/src/pages/createLinkPage/createLinkPage.tsx:14]: do you wanna write the stubs for the e2e tests? one step towards better practices lol

Comment (rikhilrai) [reply app/src/pages/createLinkPage/createLinkPage.tsx:14]: i'm planning on filling out the login ones soon

File: app/src/pages/createLinkPage/createLinkPage.tsx (hunk 1 above)
Comment (krishan711) [reply app/src/pages/createLinkPage/createLinkPage.tsx:60]: ha sorry copy-pasta

File: app/src/pages/createLinkPage/createLinkPage.tsx (hunk 8 above)
Comment (krishan711) [reply app/src/pages/createLinkPage/createLinkPage.tsx:14]: added

**Review by rikhilrai — APPROVED** [review]

**Pull request merged**